### PR TITLE
ddl: abstract decode job args v1

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #53930 

Problem Summary:

### What changed and how does it work?
add `decodeV1` to JobArgs, adjust other fillJobV1 to return args directly

### Check List

Tests 

- [x] Unit test, existing tests should cover it
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by D3Hunter** [comment]
v1 pass https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fpull_unit_test_ddlv1/detail/pull_unit_test_ddlv1/5/pipeline

File: pkg/meta/model/job_args.go [hunk 1]
```diff
@@ -1404,63 +1181,50 @@ type PlacementPolicyArgs struct {
...
-		job.Args = []any{a.Policy, a.ReplaceOnExist}
+		return []any{a.Policy, a.ReplaceOnExist}
 	} else if job.Type == ActionAlterPlacementPolicy {
-		job.Args = []any{a.Policy}
-	} else {
-		intest.Assert(job.Type == ActionDropPlacementPolicy, "Invalid job type for PlacementPolicyArgs")
-		job.Args = []any{a.PolicyName}
+		return []any{a.Policy}
 	}
+	intest.Assert(job.Type == ActionDropPlacementPolicy, "Invalid job type for PlacementPolicyArgs")
```
Comment (tangenta) [inline pkg/meta/model/job_args.go:1190]: How about returning the error instead? If `job.Type` is not one of `ActionCreatePlacementPolicy`, `ActionAlterPlacementPolicy` or `ActionDropPlacementPolicy`, it should return an error in non-test environment.

File: pkg/meta/model/job_args.go [hunk 2]
```diff
@@ -71,53 +84,59 @@ func getOrDecodeArgsV2[T JobArgs](job *Job) (T, error) {
...
+	if job.Version == JobVersion1 {
+		return getOrDecodeArgsV1[T](args, job)
+	}
+	return getOrDecodeArgsV2[T](job)
+}
+
 // JobArgs is the interface for job arguments.
 type JobArgs interface {
-	// fillJobV1 fills the job args v1 for submitting job. we make it private to
+	// getArgsV1 fills the job args v1 for submitting job. we make it private to
```
Comment (tangenta) [inline pkg/meta/model/job_args.go:96]:
```suggestion
	// getArgsV1 gets the job args v1 from submitting job. we make it private to
```

File: pkg/meta/model/job_args.go (hunk 1 above)
Comment (D3Hunter) [reply pkg/meta/model/job_args.go:1190]: we can remove it actually, only 3 job types share this `Args` struct

**Review by tangenta — APPROVED** [review]

**Review by joechenrh — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**